src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MemoryHelpers.cs

```diff
@@ -3,6 +3,7 @@
 
 using System;
 using System.Runtime;
+using System.Runtime.CompilerServices;
 
 namespace Internal.Runtime.CompilerHelpers
 {
```

Comment:
Can this copy be deleted now that you have added one to shared CoreLib?

Reply:
That is what my initial version did but I couldn't figure out how to deal with NAOT errors.

NAOT seems to be expecting helpers in `Internal.Runtime.CompilerHelpers` (I don't know where it's set, I know for sure it's not just `HelperExtensions.cs` - NAOT started to fail with "method compiled but was never scanned" something like that.

I can re-use the existing MemoryHelpers.cs from coreclr but that requires more changes on VM side to add a new Type + `ifdefs` in the *.cs file for namespace (or typename as well if we want to change it to partial RuntimeHelpers.cs) - sounds like code duplication is the simplest option?

Reply:
> NAOT started to fail with "method compiled but was never scanned" something like that.

That actually points to an existing implementation bug: there now needs to be code here which adds helpers as dependencies:

https://github.com/dotnet/runtime/blob/2f43856e78faebe7347fdd76acc26de7ffaec539/src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs#L1397C1-L1398C48

Reply:
> That actually points to an existing implementation bug: there now needs to be code here which adds helpers as dependencies:

I am not sure I understand, should I remove it from there? If I do I get 

```
*** Methods compiled but not scanned:
[S.P.CoreLib]System.Runtime.CompilerServices.RuntimeHelpers.MemSet(uint8&,uint8,native uint)
[S.P.CoreLib]System.Runtime.CompilerServices.RuntimeHelpers.MemCopy(uint8&,uint8&,native uint)
Unhandled exception: System.Exception: Scanning failure
```

Reply:
> I am not sure I understand, should I remove it from there?

The bug is that they are not present. The code should read something like:
```
void ImportInitBlk()
{
    _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.MemSet), ...);
}
```
But it won't be enough (see my edit above). Not sure what's the best way to deal with that - perhaps just always add them to the scanner's dependency graph.

Reply:
Thanks! Addressed via https://github.com/dotnet/runtime/pull/98623/commits/0d4f4f4b95b6e5e79b92c5a3dc04cb3821c639ef